src/types/server/api/user.d.ts

```diff
@@ -0,0 +1,31 @@
+type OptionalLongUrlProperty = {
+  longUrl?: string
+}
+
+type UserIdProperty = {
+  userId: number
+}
+
+type ShortUrlProperty = {
+  shortUrl: string
+}
+
+type ShortUrlOperationProperty = UserIdProperty & ShortUrlProperty
+
+type NewUserEmailProperty = {
+  newUserEmail: string
+}
+
+type StateProperty = {
+  state: 'ACTIVE' | 'INACTIVE'
+}
+
+export type UrlCreationRequest = ShortUrlOperationProperty &
```

Comment:
Nitpicking but would it be more appropriate to name the types as `RequestBody` rather than just `Request` so that it is less ambiguous?

Reply:
Makes sense. I'm following the naming convention set in [handlers.d.ts](https://github.com/opengovsg/GoGovSG/blob/73b9ae86b80df2befd4c7a13bd95162c84697eee/src/types/server/api/login/handlers.d.ts#L9) and hence just using `Request` here. Perhaps it's worth checking this code in first and then do another separate PR to refactor everything all at once?